### ISSUE TITLE
Doesn't format well if a table is nested under a bullet

### ISSUE THREAD

**Issue by dummifiedme** [tgrosinger/advanced-tables-obsidian#43]
When having a nested table (to have table indented inside a bullet), the "prettiness" seems to have lost. It fails to have that nice format. 

Can something be done about it from your side or is it an obsidian/markdown issue?

Thanks for the plugin :))

**Comment by tgrosinger** [comment]
Can you please provide a screenshot, or even better, copy the text that demonstrates the problem? (In github, paste the text inside a code block using three ` characters).

**Comment by dummifiedme** [comment]
Heres an example:

![image](https://user-images.githubusercontent.com/68785729/102965801-9700b300-4514-11eb-8d25-6db23f1f25d1.png)
![image](https://user-images.githubusercontent.com/68785729/102965811-9cf69400-4514-11eb-9503-311135905f40.png)

Any table that is nested, doesn't show any "advanced formatting'.

**Comment by tgrosinger** [comment]
Hi @dummifiedme, 
I do not believe that the last three tables in your example are valid markdown. It's tricky because there is no single markdown spec. But generally my understanding is that tables need to be on their own line. To write a table nested in a list, I would recommend using something like this:

```markdown
- Bullet 1

    | One | Two |
    | --- | --- |
    | 1   | 2   |
    |     |     |

- Bullet 2
```

It is true however that this is also not recognized by this plugin. I will look into fixing that.

**Comment by dummifiedme** [comment]
> It is true however that this is also not recognized by this plugin. I will look into fixing that.

Yes this would be fine as well :)
I basically tried to put every case there could be.

**Comment by tgrosinger** [comment]
I did a little more research into this. It looks like the plugin does function in every way when the table is indented as I showed above except that the monospace font does not work.

Unfortunately, because of how CodeMirror displays text in edit mode, I have no method of detecting tables which are indented. It might be possible to do custom parsing and add additioonal classes to lines, but then my code would be fighting with CodeMirror and I don't think it would end up very stable.

So unfortunately this is "can't completely fix" for now. But like I said, the functionality works, it just doesn't look as nice in edit mode.

**Comment by dummifiedme** [comment]
Sure, thanks for looking into it :)
And many thanks for the plugin, its really helpful

**Comment by cja-github** [comment]
There's now a new Edit mode which combines editing and viewing. How can I put my table under my bullet point?
![image](https://user-images.githubusercontent.com/5276879/185169781-0fa6f0aa-afc4-4e49-9d3d-9f71d9644c8e.png)

**Comment by asheshstha** [comment]
I think I have found a workaround. To solve the issue of failure to properly format the preview when nested in bullet point, add a tab and two spaces before first pipe of the table.
- in live preview
![image](https://github.com/tgrosinger/advanced-tables-obsidian/assets/34856377/42b89a7f-962f-4731-927d-07cd1044987f)

- in preview
![image](https://github.com/tgrosinger/advanced-tables-obsidian/assets/34856377/09355487-ea89-4f9c-93ab-30a8b64417ba)

**Comment by hmmmIndeed** [comment]
(Sorry for adding onto a dead thread)

> I think I have found a workaround. To solve the issue of failure to properly format the preview when nested in bullet point, add a tab and two spaces before first pipe of the table.
> 
>     * in live preview
>       ![image](https://github.com/tgrosinger/advanced-tables-obsidian/assets/34856377/42b89a7f-962f-4731-927d-07cd1044987f)
> 
>     * in preview
>       ![image](https://github.com/tgrosinger/advanced-tables-obsidian/assets/34856377/09355487-ea89-4f9c-93ab-30a8b64417ba)

This worked but to a degree. For some reason, the last column gets deleted and an empty column gets added in front. I was able to work around this by adding a column at the end of the table. The bottom half of each image is the same as the top half but with the extra column added. The contents of that last column don't matter since it'll be removed.

The raw markdown text in the edit view with source mode on (it looks almost the same with source mode off):

<img width="753" height="778" alt="Image" src="https://github.com/user-attachments/assets/f7ed6bb0-c01b-480b-88b7-9f3747d783eb" />

The rendered text in Obsidian reading view:

<img width="754" height="842" alt="Image" src="https://github.com/user-attachments/assets/76df870d-49cb-4072-8085-5cb31a1f3610" />

I also am unable to add the tabs and spaces before the tables with this extension enabled.

TLDR: Your tab and two spaces method works but an extra column at the end is needed to not lose the last column in reading mode. I also can't tab and space the tables with this extension enabled.

This shouldn't matter but I'm also using the [ITS Theme](https://github.com/SlRvb/Obsidian--ITS-Theme).
